server/resp/src/main/java/org/infinispan/server/resp/commands/set/SCARD.java

```diff
@@ -0,0 +1,48 @@
+package org.infinispan.server.resp.commands.set;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.multimap.impl.EmbeddedSetCache;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.util.concurrent.CompletionStages;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * {@link} https://redis.io/commands/scard/
+ *
+ * Returns the set cardinality (number of elements) of the set stored at key.
+ *
+ * @since 15.0
+ */
+public class SCARD extends RespCommand implements Resp3Command {
+   public SCARD() {
+      super(2, 1, 1, 1);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+         ChannelHandlerContext ctx,
+         List<byte[]> arguments) {
+
+      if (arguments.size() != 1) {
+         // ERROR
+         RespErrorUtil.wrongArgumentNumber(this, handler.allocator());
+      }
+      EmbeddedSetCache<byte[], WrappedByteArray> esc = handler.getEmbeddedSetCache();
+      return CompletionStages.handleAndCompose(esc.size(arguments.get(0)), (r, t) -> {
```

Comment:
Jose has improved this code so we don't need to call this way anymore. you can review other commands to see

Reply:
cleaned up